common/dns.go

```diff
@@ -0,0 +1,65 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/giantswarm/clustertest/pkg/application"
+	"github.com/giantswarm/clustertest/pkg/wait"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func runDNS() {
+	Context("dns", func() {
+		var (
+			resolver *net.Resolver
+			values   *application.DefaultAppsValues
+		)
+
+		BeforeEach(func() {
+			values = &application.DefaultAppsValues{}
+			err := Framework.MC().GetHelmValues(Cluster.Name, Cluster.Namespace, values)
+			Expect(err).NotTo(HaveOccurred())
+
+			resolver = &net.Resolver{
+				PreferGo:     true,
+				StrictErrors: true,
+				Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+					d := net.Dialer{
+						Timeout: time.Millisecond * time.Duration(10000),
+					}
+					return d.DialContext(ctx, "udp", "8.8.4.4:53")
```

Comment:
This wont work for private clusters, right? The API server endpoint wont resolve correctly.

Reply:
I think you're right, if internet traffic is only allowed through HTTP proxy then port 53 will get dropped.

Reply:
I was actually thinking more that Google DNS will return the wrong IP address for `api.xxxxx` compared to our VPN DNS server. But yeah, that also might be the case, I'm not sure if we allow DNS by default or not.

Reply:
We have an issue to add private cluster tests here https://github.com/giantswarm/roadmap/issues/2614
I suggest we look at solutions then.

Reply:
In theory the public record should be correct as it should resolve the actual IP of the cluster anyway that would only be reachable through the VPN. But then I don't know if other providers do it differently

Reply:
No, at least not on AWS. We use private Route53 which isn't available publicly. (If I'm remembering correctly). The external will return the wildcard IP which isn't the right one.

Reply:
Interesting I didn't know that, we create records in the "normal" route53.

Reply:
I think you are mixing "private hosted zone" and "public hosted zone" with the route53 api as such. The Route53 api is always publicly available to my knowledge but hosted zones can be private.